src/content/en/tools/workbox/guides/migrate.md

```diff
@@ -0,0 +1,224 @@
+
+`gulp` usage along the lines of
+
+```
+const swPrecache = require('sw-precache');
+gulp.task('generate-service-worker', function() {
+  return swPrecache.write('service-worker.js', {
+    // Config options.
+  });
+});
+```
+
+can be changed to
+
+```
+const workboxBuild = require('workbox-build');
+gulp.task('generate-service-worker', function() {
+  return workboxBuild.generateSW({
+    // Config options.
+  });
+});
+```
+
+### From the sw-precache-webpack-plugin to the Workbox webpack plugin
+
+Developers using the
+[`sw-precache-webpack-plugin`](https://www.npmjs.com/package/sw-precache-webpack-plugin) as part of
+their [webpack](https://webpack.js.org/) build process can migrate by switching to the `GenerateSW`
+class within the `workbox-webpack-plugin` module.
+
+`workbox-webpack-plugin` integrates directly with the webpack build process and "knows" about all of
+the assets generated by the a given compilation. This means that, for many use cases, you can rely
+on the default behavior of `workbox-webpack-plugin` without additional configuration, and get an
+equivalent service worker to what `sw-precache-webpack-plugin` provides.
+
+```
+const SWPrecacheWebpackPlugin = require('sw-precache-webpack-plugin');
+const webpackConfig = {
+  // ...
+  plugins: [
+    new SWPrecacheWebpackPlugin({
+        dontCacheBustUrlsMatching: /\.\w{8}\./,
+        filename: 'service-worker.js',
+      }
+    ),
+  ],
+};
+```
+
+can be changed to
+
+```
+const {GenerateSW} = require('workbox-webpack-plugin');
+const webpackConfig = {
+  // ...
+  plugins: [
+    new GenerateSW({
+      // Config options, if needed.
+    }),
+  ],
+};
+```
+
+## sw-toolbox migrations
+
+### Migrate from hand-crafted sw-toolbox usage to workbox-sw
+
+If you were using `sw-toolbox` directly (rather than using it implicity via `sw-precache`'s
+`runtimeCaching` option), then the migration to Workbox requires some manual adjustments to get the
+equivalent behavior. Reading the documentation for the
+[`workbox-routing`](/web/tools/workbox/modules/workbox-routing) and
+[`workbox-strategies`](/web/tools/workbox/modules/workbox-strategies) modules can help provide more
+context. Here are a few equivalent code snippets to help guide the migration:
+
+```
+// This sw-toolbox code:
+
+importScripts('path/to/sw-toolbox.js');
+
+// Set up a route that matches HTTP 'GET' requests.
+toolbox.router.get(
+  // Match any URL that contains 'ytimg.com', regardless of
+  // where in the URL that match occurs.
+  /\.ytimg\.com\//,
+
+  // Apply a cache-first strategy to anything that matches.
+  toolbox.cacheFirst,
+
+  {
+    // Configure a custom cache name and expiration policy.
+    cache: {
+      name: 'youtube-thumbnails',
+      maxEntries: 10,
+      maxAgeSeconds: 30
+    }
+  }
+);
+
+// Set a default network-first strategy to use when
+// there is no explcit matching route:
+toolbox.router.default = toolbox.networkFirst;
+```
+
+```
+// ...is equivalent to this Workbox code:
```

Comment:
Neither should this. 

These are two different pieces of code. They should look that way to the eye.